src/temp/ProjectImagesSlider/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { ReactElement, useEffect } from 'react';
+import Stories from 'react-insta-stories';
+import getImageUrl from '../../utils/getImageURL';
+import { SliderImage } from '../../features/projects/components/PlantLocation/ImageSlider';
+import { ProjectSingleImage } from './ProjectSingleImage';
+import styles from './Slider.module.scss';
+
+interface Props {
+  images: SliderImage[];
+  height: number | string;
+  imageSize: string;
+  type: string;
+}
+
+const ProjectImagesSlider = ({ images, height, imageSize, type }: Props) => {
+  const [slider, setSlider] = React.useState<ReactElement>();
+  const projectImages: { content: () => ReactElement }[] = [];
+
+  useEffect(() => {
+    images.forEach((sliderImage) => {
+      if (sliderImage.image) {
+        const imageURL = getImageUrl(type, imageSize, sliderImage.image);
+
+        projectImages.push({
+          content: () => (
+            <ProjectSingleImage
+              type={type}
+              imageURL={imageURL}
+              sliderImage={sliderImage}
+            />
+          ),
+        });
+      }
+    });
+    setSlider(
+      <Stories
+        stories={projectImages}
+        defaultInterval={7000}
+        width={'100%'}
+        height={height}
+        loop={true}
+        progressContainerStyles={{
+          position: 'absolute',
+          bottom: 18,
+          right: 18,
+          left: 18,
+          padding: '7px 0 5px 0',
+          maxWidth: 250,
+        }}
+        progressStyles={{ background: '#27AE60', height: 3.35 }}
+        progressWrapperStyles={{
+          height: 3.35,
+          background: 'rgba(255, 255, 255, 0.50)',
+        }}
+        storyContainerStyles={{ borderRadius: 13 }}
+      />
+    );
```

Comment:
For my understanding, is there a reason to use `setSlider` and maintain this in state, instead of the `projectImages` array?

Reply:
Yes, both ways will provide the functionality but we'd need to create a state for `projectImages` as well and initialize it with a null value. The approach in this file have been copied from the current implementation on production and so there isn't any specific reason to go with this one.

Reply:
Let's stick with this then.